src/tiled/varianteditorfactory.h

```diff
@@ -69,12 +69,18 @@ class VariantEditorFactory : public QtVariantEditorFactory
     void fileEditFileUrlChanged(const QUrl &value);
     void textPropertyEditTextChanged(const QString &value);
     void comboBoxPropertyEditTextChanged(const QString &value);
+    void enumPropEditTextChanged(const QString &value);
     void objectRefEditValueChanged(const DisplayObjectRef &value);
     void slotEditorDestroyed(QObject *object);
 
     QMap<QtProperty *, QList<FileEdit *> > mCreatedFileEdits;
     QMap<FileEdit *, QtProperty *> mFileEditToProperty;
 
+    // for now there are no other customprops.
+    QMap<QtProperty *, QList<QComboBox *> > mCreatedEnumProps;
+    QMap<QComboBox *, QtProperty *> mEnumPropToProperty;
```

Comment:
It would be nice if we could delegate widget creation somehow, because a custom type could indeed be more than just an enum but could also be an integer value with custom range for example. The `QtVariantEditorFactory` can already do this, with the only problem that its value needs to be the actual value and not a `CustomValue`, but we need to pass around a `CustomValue` in order to keep track its custom type. Making this possible would require some changes to the `qtpropertybrowser` solution.

Reply:
I've resolved this issue by moving the translation from `PropertyValue` to the associated property browser type and attributes to a `CustomPropertiesHelper`, which is used by `PropertyBrowser` and `ObjectTypesEditor` to create the properties and convert the values.